code/modules/events/disease_outbreak.dm

```diff
@@ -267,25 +274,35 @@
 				/datum/symptom/fire,
 				/datum/symptom/flesh_death,
 				/datum/symptom/flesh_eating,
-				/datum/symptom/narcolepsy,
-				/datum/symptom/voice_change,
+				/datum/symptom/visionloss,
 			)
 
-	for(var/i in 1 to max_symptoms)
+	var/current_severity = 0
+
+	while(symptoms.len < max_symptoms)
 		var/datum/symptom/chosen_symptom = pick_n_take(possible_symptoms)
-		if(chosen_symptom)
-			var/datum/symptom/new_symptom = new chosen_symptom
-			symptoms += new_symptom
 
-	if(prob(50))
-		symptoms += new /datum/symptom/viralevolution
+		if(!chosen_symptom)
+			CRASH("Advanced disease could not pick a symptom!")
+
+		var/datum/symptom/new_symptom = new chosen_symptom
+
+		if(symptoms.len == (max_symptoms - 1) && (current_severity < requested_severity) && (new_symptom.severity < requested_severity))
+			continue
+
+		symptoms += new_symptom
+		if(new_symptom.severity > current_severity)
+			name = "[new_symptom.illness]"
+			current_severity = new_symptom.severity
 
 	if(prob(50))
 		symptoms += new /datum/symptom/youth
+		name = "Advanced [name]"
 
-	Refresh()
+	if(prob(50))
+		symptoms += new /datum/symptom/viralevolution
```

Comment:
Shouldn't making it an advanced virus be on the prob(50) that we apply viral evolution on instead of the youth one, or do i misunderstand?

Reply:
50% chance of getting that particular trait. It can have both as well.